eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/session/send/UpStreamMsgContext.java

```diff
@@ -92,21 +92,26 @@ public void retry() {
             Command replyCmd = getReplyCmd(header.getCmd());
             long sendTime = System.currentTimeMillis();
 
-            EventMeshTcpSendResult sendStatus = session.upstreamMsg(header, event,
-                    createSendCallback(replyCmd, taskExecuteTime, event), startTime, taskExecuteTime);
+            retryTimes++;
 
-            if (StringUtils.equals(EventMeshTcpSendStatus.SUCCESS.name(), sendStatus.getSendStatus().name())) {
-                logger.info("pkg|eventMesh2mq|cmd={}|event={}|user={}|wait={}ms|cost={}ms", header.getCmd(), event,
+            // check session availability
+            if (session.isAvailable(event.getSubject())) {
```

Comment:
If so, then isAvailable needs to be more refined. This modification is because the session has been closed, the retry logic is still calling session.upstreamMsg, an unnecessary exception occurs, and even an infinite loop.

Reply:
> If so, then isAvailable needs to be more refined. This modification is because the session has been closed, the retry logic is still calling session.upstreamMsg, an unnecessary exception occurs, and even an infinite loop.

If so, then isAvailable needs to be more refined. This modification is because the session has been closed, the retry logic is still calling session.upstreamMsg, an unnecessary exception occurs, and even an infinite loop.

Reply:
> session.isAvailable() is not suitable for use here，this method was originally used to determine whether a message can be sent to a consumer session. The method contains logic as follows:
> 
> ```
>       if (!sessionContext.subscribeTopics.containsKey(topic)) {
>           logger.warn("session is not available because session has not subscribe topic:{},client:{}", topic, client);
>           return false;
>       }
> ```

If so, then isAvailable needs to be more refined. This modification is because the session has been closed, the retry logic is still calling session.upstreamMsg, an unnecessary exception occurs, and even an infinite loop.

Reply:
Yes, that's right. You can supply another method for replacing`isAvailable`.

Reply:
> Yes, that's right. You can supply another method for replacing`isAvailable`.

Ok, it's done.